### DISCUSSION TITLE
Partial Render

### DISCUSSION THREAD

**Discussion by cemrehancavdar** [jpsca/jinjax#62] [General]
Is it possible to partially render JinjaX templates?

**Comment by jpsca** [comment]
Of course, you don;t have to render a full page, you can render _any_ component with `catalog.irender("MyComponent", **context)` or `catalog.render("MyComponent", **context)` .

The difference between the two is that  `catalog.render` reset the `catalog.collected_css` and `catalog.collected_js` lists that `catalog.render_assets()` uses, while `catalog.irender` doesn't. So:

1. if you are rendering some components to insert them into another template, you should use `irender`, so you can also do something like `html_assets = catalog.render_assets()`, at the end, to send it to the external template.
2. If you want just to re-render a subcomponent to do a partial response (for instance,  when using htmx or turbo), it doesn't matter.
3. If you are rendering a full page, use `render` so only the assets from the page are rendered.
